metricsx/bridge_segment.go

```diff
@@ -0,0 +1,238 @@
+package metricsx
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"runtime"
+	"time"
+
+	"github.com/pborman/uuid"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/segmentio/analytics-go"
+	"github.com/sirupsen/logrus"
+
+	"github.com/ory/x/resilience"
+)
+
+type analyticsProperties struct {
+	memstats analytics.Properties
+	app      analytics.Properties
+}
+
+// This map is for converting Prometheus metrics (by name) to segment keys that were
+// previously sent from ory/x/metricsx
+var memStats = map[string]string{
```

Comment:
Reading concurrently from maps can cause panics as it is not specified by the language definition that maps are thread safe at all.